storage/local/storage.go

```diff
@@ -212,10 +212,18 @@ func (s *memorySeriesStorage) Start() error {
 		return err
 	}
 	s.persistence = p
+	go s.persistence.run()
+
+	pclose := func() {
```

Comment:
You could have a `defer` here that checks for the value of err and closes the persistence if err != nil.